apollo-plugin/apollo-log4j2/src/main/java/com/ctrip/framework/apollo/log4j2/ApolloLog4j2ConfigurationFactory.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2022 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.ctrip.framework.apollo.log4j2;
+
+import com.ctrip.framework.apollo.ConfigFile;
+import com.ctrip.framework.apollo.ConfigService;
+import com.ctrip.framework.apollo.core.enums.ConfigFileFormat;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.ConfigurationException;
+import org.apache.logging.log4j.core.config.ConfigurationFactory;
+import org.apache.logging.log4j.core.config.ConfigurationSource;
+import org.apache.logging.log4j.core.config.Order;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.xml.XmlConfiguration;
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * @author nisiyong
+ */
+@Plugin(name = "ApolloLog4j2ConfigurationFactory", category = ConfigurationFactory.CATEGORY)
+@Order(50)
+public class ApolloLog4j2ConfigurationFactory extends ConfigurationFactory {
```

Comment:
> Does this plugin only use with apollo-client?

Yes.

> Maybe we can encode the `client` to this class's name and the code, for gain more accurate naming?

Could you explain more?

Reply:
It seems that the comment lossed.

i.e maybe rename ApolloLog4j2ConfigurationFactory to ApolloClientLog4j2ConfigurationFactory is better? Because this ConfigurationFactory only use in apollo-client?

Reply:
`ApolloClient*` is acceptable to me, and I think `ApolloClientConfigurationFactory` is better. The name has been changed.